src/sonic_ax_impl/mibs/ietf/rfc4292.py

```diff
@@ -49,26 +50,34 @@ def update_data(self):
             self.route_dest_list.append(sub_id)
             self.route_dest_map[sub_id] = self.loips[loip].packed
 
-        route_entries = Namespace.dbs_keys(self.db_conn, mibs.APPL_DB, "ROUTE_TABLE:*")
-        if not route_entries:
-            return
-
-        for route_entry in route_entries:
-            routestr = route_entry
-            ipnstr = routestr[len("ROUTE_TABLE:"):]
-            if ipnstr == "0.0.0.0/0":
-                ipn = ipaddress.ip_network(ipnstr)
-                ent = Namespace.dbs_get_all(self.db_conn, mibs.APPL_DB, routestr, blocking=True)
-                nexthops = ent["nexthop"]
-                ifnames = ent["ifname"]
-                for nh, ifn in zip(nexthops.split(','), ifnames.split(',')):
-                    ## Ignore non front panel interfaces
-                    ## TODO: non front panel interfaces should not be in APPL_DB at very beginning
-                    ## This is to workaround the bug in current sonic-swss implementation
-                    if ifn == "eth0" or ifn == "lo" or ifn == "docker0": continue
-                    sub_id = ip2tuple_v4(ipn.network_address) + ip2tuple_v4(ipn.netmask) + (self.tos,) + ip2tuple_v4(nh)
-                    self.route_dest_list.append(sub_id)
-                    self.route_dest_map[sub_id] = ipn.network_address.packed
+        # Get list of front end asic namespaces for multi-asic platform.
+        # This list will be empty for single asic platform.
+        front_ns = multi_asic.get_all_namespaces()['front_ns']
+
+        for db_conn in Namespace.get_non_host_dbs(self.db_conn):
+            # For multi-asic platform, proceed to get routes only for front end namespaces.
+            if db_conn.namespace != multi_asic.DEFAULT_NAMESPACE and \
+               db_conn.namespace not in front_ns:
+                continue
+            ipnstr = "0.0.0.0/0"
+            route_str = "ROUTE_TABLE:0.0.0.0/0"
+            route_entry  = db_conn.keys(mibs.APPL_DB, route_str)
+            if not route_entry:
+                continue
+            ipn = ipaddress.ip_network(ipnstr)
+            ent = db_conn.get_all(mibs.APPL_DB, route_str, blocking=True)
+            nexthops = ent["nexthop"]
+            ifnames = ent["ifname"]
+            for nh, ifn in zip(nexthops.split(','), ifnames.split(',')):
+                ## Ignore non front panel interfaces
+                ## TODO: non front panel interfaces should not be in APPL_DB at very beginning
+                ## This is to workaround the bug in current sonic-swss implementation
+                if ifn == "eth0" or ifn == "lo" or ifn == "docker0": continue
+                # Ignore internal asic routes
+                if multi_asic.is_port_channel_internal(ifn, db_conn.namespace): continue
```

Comment:
check for backend ports also to be on safer side.

Reply:
updated as per comment. could not make use of multi_asic.is_port_internal() directly, because if we pass a port channel name to this function, then it will throw an error : Unknown port name . So instead checking the port_config table directly.